core/unit_test/TestAbort.hpp

```diff
@@ -59,8 +59,10 @@ template <class ExecutionSpace>
 struct TestAbortPrintingToStdout {
   TestAbortPrintingToStdout() {
     ::testing::internal::CaptureStdout();
-    Kokkos::parallel_for(Kokkos::RangePolicy<ExecutionSpace>(0, 1), *this);
-    Kokkos::fence();
+    ExecutionSpace exec;
+    Kokkos::parallel_for(Kokkos::RangePolicy<ExecutionSpace>(exec, 0, 1),
+                         *this);
+    exec.fence();
```

Comment:
I'm still not quite sure why the `HIP` Ci was timing out without changing this and I'm not quite sure how much we care. 
I couldn't reproduce on `crusher` or `spock`.

Reply:
Also, @JBludau can only reproduce with ROCm < 5.

Reply:
So you are saying this is not hanging with 5.2?